capstone/src/test/java/com/google/step/servlets/RestaurantServletTest.java

```diff
@@ -0,0 +1,187 @@
+package com.google.step.servlets;
+
+import static org.junit.Assert.*;
+
+import com.google.appengine.api.datastore.*;
+
+import com.meterware.httpunit.*;
+import com.meterware.servletunit.ServletRunner;
+
+import com.meterware.servletunit.ServletUnitClient;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import org.xml.sax.SAXException;
+
+import java.io.IOException;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class RestaurantServletTest {
+    private final LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    /**
+     * Test for the doGet method of the RestaurantServlet. First adds
+     * a dummy entity to the mock Datastore then calls doGet using a
+     * ServletRunner to check the response against the expected format.
+     * @throws IOException
+     * @throws SAXException if the ServletRunner parser cannot handle the request
+     */
+    @Test
+    public void testDoGet() throws IOException, SAXException {
+        DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+
+        // Create dummy RestaurantInfo entity and add it to the mock Datastore
+        Entity restaurantInfo1 = new Entity("RestaurantInfo");
+        restaurantInfo1.setProperty("restaurantKey", 4);
+        restaurantInfo1.setProperty("name", "Wildfire");
+        restaurantInfo1.setProperty("location", new GeoPt((float) 42.17844, (float) -87.92843));
+        restaurantInfo1.setProperty(
+                "story", "Swanky American chain serving steak, chops & seafood, plus burgers, sides & cocktails.");
+        List<String> cuisineList1 = Arrays.asList("Steakhouse", "American");
+        restaurantInfo1.setProperty("cuisine", cuisineList1);
+        restaurantInfo1.setProperty("phone", "(847)279-7900");
+        restaurantInfo1.setProperty("website",
+                "https://www.wildfirerestaurant.com");
+        restaurantInfo1.setProperty("score", 1.1);
+        restaurantInfo1.setProperty("status", "STRUGGLING");
+        datastoreService.put(restaurantInfo1);
+
+        // Set expected JSON response based on props
+        String expected = "{\"restaurant\":\"{\\\"value\\\":{\\\"restaurantKey\\\":4,\\\"name\\\":\\\"Wildfire\\\"," +
```

Comment:
It might be less of a headache if you defined this as a JSON object, and then parsed the output from the actual response, and did the comparison there.

Also, this might be broken, because you're relying on the key ordering of a map being consistent, which is not guaranteed. Note that in the golang language specifically, they go out of their way to randomize iteration order, JUST to mess with this type of practice. :)